design/STREAM_STATE_DDB_PROPOSAL.md

```diff
@@ -0,0 +1,110 @@
+# Fix Collection States Proposal
+
+## Problem
+
+When an invocation for a collector repeatedly fails for a period longer than the lifetime of an SQS message, 
+the collection state stored in that message is permanently lost. As a result, collection effectively stops 
+for that stream. Should this issue occur across all streams in the collector, such as a bad credential or OOM,
+then all collection for the collector ceases, and only healthchecks are left. 
+
+## Solution
+
+Implement a script that is run periodically that shall check if collection has ceased for a given collector by 
+checking against the collector state database to ensure that collection has occured recently.
```

Comment:
I think we need to gracefully handle all errors (perhaps on paws lib level) and make sure collection state message never expires in SQS (by resubmitting it).